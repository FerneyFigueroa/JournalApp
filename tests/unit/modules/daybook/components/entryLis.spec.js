import { shallowMount } from "@vue/test-utils"
import { createStore } from "vuex"

import journal from '@/modules/daybook/store/journal'

import EntryList from '../../../../../src/modules/daybook/components/EntryList.vue'

import {journalState} from '../../../mock-date/test-journal-state'

const createVuexStore  =  (inicialState)=> 
createStore({
    modules:{
        journal:{
            ...journal,
            state:{...inicialState}
        }
    }
})


describe('pruebas en el entry list',()=>{
   const store = createVuexStore(journalState)
   const mockRouter = {
     push: jest.fn()
   }
    
   let wrapper 
    beforeEach(()=>{
        jest.clearAllMocks()
        wrapper = shallowMount(EntryList,{
            global:{
                mocks:{
                     $router: mockRouter
                },
                plugins:[store]
            }
        })
    })

    test('debe de llamar el getEntriesByTer sin termino y mostrar dos entrads',()=>{

        expect(wrapper.findAll('entry-stub').length).toBe(0)
        expect(wrapper.html()).toMatchSnapshot()

    })

    test('debe de  llamar el getentriesterm y filtrar las entradas',async()=>{
        const input = wrapper.find('input')
        await input.setValue('segunda')

        expect(wrapper.findAll('entry-stub').length).toBe(0)
    })

    test('el boton de nuevo debe de redireccionar a /new',()=>{
        wrapper.find('button').trigger('click')

        expect(mockRouter.push)
            .toHaveBeenCalledWith({name:'entry',params:{id:'new'}})
    })
})