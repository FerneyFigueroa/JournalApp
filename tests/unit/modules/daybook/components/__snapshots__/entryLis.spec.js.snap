// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el entry list debe de llamar el getEntriesByTer sin termino y mostrar dos entrads 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="
        text" class="form-control" placeholder="Buscar Entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva Entrada </button></div>
  <div class="entry-scrollarea">
    <entry-mi-stub entry="[object Object]"></entry-mi-stub>
  </div>
</div>
`;
